core/src/main/java/com/linecorp/armeria/client/HttpStatusClassPredicates.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Predicate;
+
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.HttpStatusClass;
+
+final class HttpStatusClassPredicates implements Predicate<HttpStatus> {
```

Comment:
I wasn't sure how this predicate plays into the current APIs as nothing is public 😅 
I'm also curious how the `HttpStatus[Class]Predicate`s tie into the new APIs since. Can you show me an example?

Reply:
I think the same too.
I'm not sure how to use `HttpStatus[Class]Predicate` in the new APIs.
It would be unnecessary so that I would remove this class.